### PR TITLE
Merge Skip Device Slider Potentiometers Into Skip Device Main

### LINKED ISSUE #1: ADC Drivers for Potentiometers
Write the drivers for the skip ADC potentiometers.

### PR DESCRIPTION
**Pull request by GrahamDrive**
I finished the slider driver. 

It uses a struct called slider_t that contains all the information necessary for the moving average filter and the pin number of the given slider it can be seen in the second image.

```c++
typedef struct{
    uint8_t index = 0;            // Index of the moving average
    uint16_t data[RAWindowSize];  // Data from rolling average
    uint32_t avgSum = 0;          // Sum of values in data
    uint16_t pin = INT16_MAX;     // Pin of the slider
}slider_t;
``` 

Below you can see a couple functions, the pollSlider function is intended to be called in the main loop of the program in order to constantly update the rolling average of the potentiometers. The others are get functions that return the averaged ADC value of the sliders and the function returns the average value in a percentage format.
```c++
SLIDE_POT_StatusTypeDef pollSlider(slider_t *slider){
    SLIDE_POT_StatusTypeDef status = SLIDE_POT_OK;
    uint16_t potValue;

    //assert(slider.pin != NULL); Some sort of error checking should be used in future.

    potValue = analogRead(slider->pin);

    // Moving Average Code
    slider->avgSum -= slider->data[slider->index]; // Remove oldest value
    slider->avgSum += potValue; // Add new value 
    slider->data[slider->index] = potValue; // Set position to new value
    slider->index++; // Increase index
    slider->index %= RAWindowSize; // If index larger than window set to 1

    error:
        return status;
}


uint16_t getAveragePotValue(slider_t *slider){
    return (uint16_t)(slider->avgSum / RAWindowSize);
}

uint8_t getPercentagePotValue(slider_t *slider){
    return (uint8_t)((float)100 * ( (float)slider->avgSum / (float)RAWindowSize)/ (float)4096);
}
```

I also added some setup code and a section of how it would run into main. We can just add other pieces of code in main around in their own sections similar to what I did.

### CODE REVIEW

**Review by AlexRHaigh — APPROVED** [review]
Looks Good!

File: Skip-Device/src/main.cpp [hunk 1]
```diff
@@ -1,18 +1,53 @@
...
 void loop() {
-  // put your main code here, to run repeatedly:
-}
+  // main loop
+
+
+  // Slider polling Section
+  // Make sure sliders are constantly polled
+  pollSlider(&leftSlider);
+  pollSlider(&rightSlider);
```
Comment (RiAllen120) [inline Skip-Device/src/main.cpp:42]: Shouldn't this be in the if statement as well so it doesn't poll with the clock frequency?

**Review by RiAllen120 — APPROVED** [review]
We can overview the power consumption to continuously poll the sliders at a later timer, otherwise it looks good

File: Skip-Device/src/main.cpp (hunk 1 above)
Comment (GrahamDrive) [reply Skip-Device/src/main.cpp:42]: Yeah we could I just would opt for getting things working at full power then we can work on power consumption.

**Pull request merged**